aws-android-sdk-appsync/src/test/java/com/amazonaws/mobileconnectors/appsync/AppSyncClientUnitTest.java

```diff
@@ -252,6 +252,11 @@ public void onFailure(@Nonnull ApolloException e) {
                 waitForCall.countDown();
             }
 
+            @Override
+            public void onCreated() {
+
+            }
```

Comment:
You should add a test that it gets called.

```suggestion
            @Override
            public void onCreated() {
            }
```

Reply:
Hey @jamesonwilliams thanks for this feedback - I hadn't considered that backwards compatibility would be a concern 🤦 

I have just pushed a change to introduce a new interface that extends `AppSyncSubscriptionCall.Callback` - that way users can opt-in to this feature. I'm only 2% satisfied with the name of the new interface so if you have a better suggestion then please share!

As to the unit test, I can't find a way to inject a `SUBSCRIPTION_ACK` because everything is so private! Any suggestions for how I might proceed?

Reply:
Hey @majsty, cool! This seems like a creative and workable solution.

So if I have:
```java
final class FullCallback<T> implements Callback<T>, CreationCallback<T> {
    ...
}
```
I'll get the new behavior. But existing customers who have:
```java
final class OriginalCallback<T> implements Callback<T> {
    ...
}
```
will continue to work as is.

Let me look into the unit test and see if I have any ideas.

Reply:
Ah, right. The `WebSocketConnectionManager` is difficult to unit test, since we have no way mock the WebSocket. _Theoretically_, one could use OkHttp's `MockWebServer` for this purpose. But I've burned at least a few hours before trying to figure out how to make it work with WebSockets. As https://github.com/square/okhttp/issues/2931 implies, it's not "easy." So anyway, let's punt on the unit tests for now. The current change set looks isolated and straight-forward enough.

Reply:
Thanks for looking into the test options anyway - good to confirm I wasn't missing a straightforward way of invoking it!

Your above example isn't quite how I implemented `CreationCallback` - I did it as an extension of `Callback`, so you'd only implement one or the other, not both. Would you prefer that I change it to match your suggestion?

Reply:
Oops, sorry -- I see now, yes! No, your way is better.